homeassistant/components/climate/generic_thermostat.py

```diff
@@ -287,15 +287,27 @@ def _async_control_heating(self):
                 if too_hot:
                     _LOGGER.info('Turning off heater %s',
                                  self.heater_entity_id)
-                    switch.async_turn_off(self.hass, self.heater_entity_id)
+                    self._heater_turn_off()
             else:
                 too_cold = self._target_temp - self._cur_temp >= \
                     self._cold_tolerance
                 if too_cold:
                     _LOGGER.info('Turning on heater %s', self.heater_entity_id)
-                    switch.async_turn_on(self.hass, self.heater_entity_id)
+                    self._heater_turn_on()
 
     @property
     def _is_device_active(self):
         """If the toggleable device is currently active."""
-        return switch.is_on(self.hass, self.heater_entity_id)
+        return self.hass.states.is_state(self.heater_entity_id, STATE_ON)
+
+    def _heater_turn_on(self):
```

Comment:
I think you should mark this (and `_heater_turn_off`) a callback, since you call them from callbacks, i.e., they are going to be run in the event loop. They don't do anything blocking, so that should be okay.